ksa_compliance/ksa_compliance/doctype/sales_invoice_additional_fields/sales_invoice_additional_fields.py

```diff
@@ -340,17 +340,14 @@ def add_integration_log_document(self, zatca_message, integration_status, zatca_
     @property
     def qr_image(self):
         if self.qr_code:
-            qr = qrcode.QRCode(version=1, box_size=10, border=5)
-            qr.add_data(self.qr_code)
-            qr.make(fit=True)
-            img = qr.make_image(fill_color="black", back_color="white")
-            img_buffer = BytesIO()
-            img.save(img_buffer, format='PNG')
-            img_buffer.seek(0)
-            img_str = base64.b64encode(img_buffer.read()).decode("utf-8")
+            qr = pyqrcode.create(self.qr_code)
+            buffer = BytesIO()
+            qr.png(buffer, scale=7)
+            buffer.seek(0)
+            img_str = base64.b64encode(buffer.getvalue()).decode("utf-8")
             return img_str
         else:
-            return "No QR image"
+            return "No QR Code"
```

Comment:
How is the virtual field intended to be used? If the caller embeds its result directly into an `<img>` element, it won't work when it says "No QR Code". We have a couple of options:
* Return `None` when there's no QR code. The caller has to check for truthiness before embedding in an image (i.e. `if doc.qr_code: do_something`)
* Always return an image, but use a special image to indicate no QR code (e.g. load from assets or render an SVG)

I think the first option's easier, especially if we're going to just embed them in print formats.